rfc/0008-builtin-blocktypes/README.md

```diff
@@ -0,0 +1,212 @@
+
+    output oftype File;
+}
+
+builtin blocktype ArchiveInterpreter {
+    input oftype File;
+
+    property archiveType oftype text;
+
+    output oftype FileSystem;
+}
+
+builtin blocktype FilePicker {
+    input oftype FileSystem;
+
+    property path oftype text;
+
+    output oftype File;
+}
+
+builtin blocktype TextFileInterpreter {
+    input oftype File;
+
+    property encoding oftype text;
+    property lineBreak oftype regex;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextLineDeleter {
+    input oftype TextFile;
+
+    property lines oftype collection<integer>;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextRangeSelector {
+    input oftype TextFile;
+
+    property lineFrom oftype integer;
+    property lineTo oftype integer;
+
+    output oftype TextFile;
+}
+
+builtin blocktype CSVInterpreter {
+    input oftype TextFile;
+
+    property delimiter oftype text: ",";
+    property enclosing oftype text: "";
+    property enclosingEscape oftype text: "";
+
+    output oftype Sheet;
+}
+
+builtin blocktype CellRangeSelector {
+    input oftype Sheet;
+
+    property select oftype cellrange;
+
+    output oftype Sheet;
+}
+
+builtin blocktype CellWriter {
+    input oftype Sheet;
+
+    property write oftype text;
+    property at oftype cell;
+
+    output oftype Sheet;
+}
+
+builtin blocktype ColumnDeleter {
+    input oftype Sheet;
+
+    property delete oftype collection<column>;
+
+    output oftype Sheet;
+}
+
+builtin blocktype RowDeleter {
+    input oftype Sheet;
+
+    property delete oftype collection<row>;
+
+    output oftype Sheet;
+}
+
+builtin blocktype TableInterpreter {
+input oftype Sheet;
+
+    property header oftype boolean;
+    property columns oftype collection<valuetype-assignment>;
+
+    output oftype Table;
+}
+
+builtin blocktype SQLiteLoader {
+    input oftype Table;
+
+    property table oftype text;
+    property file oftype text;
+}
+
+builtin blocktype PostgresLoader {
+    input oftype Table;
+
+    property host oftype text;
+    property port oftype integer;
+    property username oftype text;
+    property password oftype text;
+    property database oftype text;
+    property table oftype text;
+}
+```
+</details>
+
+## Drawbacks
+
+- Adds valuetype keywords for properties, but these types cannot be assigned to table columns 
+
+## Alternatives
+
+- Not having declarations for builtin block types at all
+- No mixing of property valuetypes and column valuetypes
+- Using a less verbose syntax (e.g. omit `property` keyword or use a shorthand operator instead of `oftype` keyword)
+
+## Possible Future Changes/Enhancements
+
+- Usage of valuetypes for typing properties
+  - Constraints can then be used to validate property values
+  - Properties can be considered inputs of a block, so property values can be provided dynamically by pipes
+- Include validation semantics beyond valuetypes
+- Introduce a standard for documenting blocktypes in the code
```

Comment:
I really like this, Python has the concept of method docs as comment in the method, Eiffel has the pre and post conditions of methods in separate parts. I think we could really do something good by structuring and requiring documentation for the blocks, their inputs/outputs and examples like we have right now directly in the jayvee code.